lib/services/add_product_to_basket.rb

```diff
@@ -8,10 +8,20 @@ def initialize(params)
     end
 
     def call
-      BASKET << Basket.new(
-        product_id: product_id,
-        quantity: quantity
-        )
+      item = FetchBasketItem.new.call(product_id)
+
+      if item == nil
+        BASKET << Basket.new(
+          product_id: product_id,
+          quantity: quantity
+          )
+      else
+        increase_item_quantity(item, quantity)
+      end
+    end
+
+    def increase_item_quantity(item, quantity)
```

Comment:
this method could be declared on an item class. better not to manipulate object internal state from outside.